src/App.tsx

```diff
@@ -1,24 +1,247 @@
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => setLoading(false));
+  }
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  if (inputField.current) {
+    inputField.current.focus();
+  }
+
+  useEffect(() => {
+    loadTodos();
+    if (inputField.current) {
+      inputField.current.focus();
+    }
+  }, [tempTodo]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const deleteTodo = (postId: number) => {
+    setdeletedPostsIds(curId => [...curId, postId]);
+    postServices.deleteTodo(postId)
+      .then(() => {
+        setTodos(curTodos => curTodos.filter(t => t.id !== postId));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+        setTimeout(() => setdeletedPostsIds([]), 3000);
+      });
+  };
+
+  const deleteAll = () => {
+    todos.filter(t => t.completed)
+      .map((todo) => deleteTodo(todo.id));
+  };
+
+  function addTodos({ userId, title, completed }: Todo) {
+    if (title.length === 0) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    setInputDisabled(true);
+    setLoading(true);
+
+    postServices.addTodo({ userId, title, completed })
+      .then(nTodo => {
+        setTodos((curTodos) => [...curTodos, nTodo]);
+        setNewTodo('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setLoading(false);
+        setInputDisabled(false);
+        setTempTodo(null);
+      });
+  }
+
+  const handleSabmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newT = {
+      id: 0,
+      userId: USER_ID,
+      title: newTodo.trim(),
+      completed: false,
+    };
+
+    setTempTodo(newT);
+    addTodos(newT);
+  };
+
+  const visibleTodos = getPreparedTodos(todos, filter);
+  const onFilter = (f: string) => {
+    setFilter(f);
+  };
+
+  const onCloseError = (e: string) => {
+    setErrorMessage(e);
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div
+        className="todoapp__content"
+      >
+        <header className="todoapp__header">
+          {/* this buttons is active only if there are some active todos */}
+          <button
+            type="button"
+            className={visibleTodos.some(t => !t.completed)
+              ? 'todoapp__toggle-all active'
+              : 'todoapp__toggle-all'}
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally. Fix all cases